component/identity/src/main/java/org/gatein/portal/idm/impl/store/attribute/ExtendedAttributeManagerImpl.java

```diff
@@ -0,0 +1,54 @@
+package org.gatein.portal.idm.impl.store.attribute;
+
+import org.exoplatform.container.PortalContainer;
+import org.exoplatform.services.organization.idm.PicketLinkIDMService;
+import org.picketlink.idm.api.User;
+import org.picketlink.idm.common.exception.IdentityException;
+import org.picketlink.idm.impl.api.PasswordCredential;
+import org.picketlink.idm.impl.api.session.IdentitySessionImpl;
+import org.picketlink.idm.impl.api.session.managers.AttributesManagerImpl;
+import org.picketlink.idm.impl.credential.DatabaseReadingSaltEncoder;
+
+import java.util.Map;
+
+public class ExtendedAttributeManagerImpl extends AttributesManagerImpl {
+
+  public static final String          OLD_PASSWORD_SALT_USER_ATTRIBUTE = "passwordSalt";
+
+  private final Map<String, String>   oldCredentialEncoderProps        =
+                                                                Map.of("credentialEncoder.hashAlgorithm",
+                                                                       "SHA-256",
+                                                                       "credentialEncoder.secureRandomAlgorithm",
+                                                                       "SHA1PRNG",
+                                                                       "credentialEncoder.class",
+                                                                       "org.picketlink.idm.impl.credential.DatabaseReadingSaltEncoder");
+
+  private static PicketLinkIDMService idmService;
+
+  public ExtendedAttributeManagerImpl(IdentitySessionImpl session) {
+    super(session);
+  }
+
+  public boolean validatePassword(User user, String password) throws IdentityException {
+    DatabaseReadingSaltEncoder oldCredentialEncoder = new DatabaseReadingSaltEncoder();
```

Comment:
The pseudo code in the tech study is :
`check if user have attribute passwordSalt128
if yes : use the current way to check password
it will get the default configured PasswordEncoder (Argon2IdPasswordEncoder)
and use it

if no, validateCredential by using the old passwordEncoder :
DatabaseReadingSaltEncoder
it will automatically use the old configured algo to validate the password
(SHA256 or MD5 in function of the configuration)
If validation is ok, call updatePassword to compute and save a hash with the new algo. During passwordUpdate, we need to ensure to remove passwordSalt `

It has the advantage to not initialize oldEncoder if not needed. 
In addition, I wish that, if possible, check the password with the correct encoder : I dont want to test with the old one, and if not work, test the new one, because this solution implies to make 2 verification, it is not performant.

By checking the attribute passwordSalt, we have the possibility to verify if the new algo must be used, or the old one.

Reply:
done